### PR TITLE
Feature1

### PR DESCRIPTION
**Pull request by m05e5**
# Ruby on Rails capstone project - Budget app

- register and log in, so that the data is private to them.
- introduce new transactions associated with a category.
- see the money spent on each category.
- implement Splash screen
- follow these design guidelines
- [video demo](https://www.loom.com/share/b6346a44993a42aea84f8cf81ecdfb9a)

### CODE REVIEW

**Review by Dipeshtwis — COMMENTED** [review]
# Status: Invalid ❌

While you have done a great job in the project I cannot proceed with the code review. Please fix the issue mentioned below and then request for code review

## Reason

- [ ] Pull Requests should have a meaningful title and short summary.

### Please make sure to submit a code review when you are done with the changes that were requested

**Don't worry you still have same number of requests left as previous.**


Cheers and Happy Coding :clap: :clap: :clap: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

**_Dipesh Kumar_**

**Comment by m05e5** [comment]
Oh yeah sorry I forgot

File: README.md [hunk 1]
```diff
@@ -1,24 +1,111 @@
...
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Microverse
+
```
Comment (Dipeshtwis) [inline README.md]: - [x] Here you should give appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -13,3 +13,61 @@
...
+  align-items: center;
+  justify-content: center;
+  height: 85vh;
+  background-color: rgb(245, 245, 245);
+}
+
+.form-container {
+  /* display: flex;
+  flex-direction: column;
+  align-items: center; */
```
Comment (Dipeshtwis) [inline app/assets/stylesheets/application.css]: - [x] I would recommend removing the commented code to make the codebase cleaner.

File: app/assets/stylesheets/entities.scss [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  position: fixed;
+  background-color: rgb(77, 77, 77);
+  z-index: 3;
+  flex-direction: column;
+  border-width: 1px;
+  border-style: solid;
+  border-color: white;
+  color: white;
+  font-size: 20px;
+  // overflow: scroll;
```
Comment (Dipeshtwis) [inline app/assets/stylesheets/entities.scss]: - [x] Here as well, I would recommend removing the commented code to make the codebase cleaner.

File: app/assets/stylesheets/entities.scss [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+.X {
+  // margin-left: auto;
+  // position: fixed;
+  // left: 55%;
+  // top: 12%;
+  cursor: pointer;
+}
+
+/* The Modal (background) */
```
Comment (Dipeshtwis) [inline app/assets/stylesheets/entities.scss]: - [x] Here as well. I would recommend removing the commented code to make the codebase cleaner.

File: app/controllers/entities_controller.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      @entities = @category.category_with_entities.order(created_at: :desc)
+      @total_amount = @category.total_amount
+
+    else
+      redirect_to(root_path, notice: 'you dont have access')
+    end
+  end
+
+  def create
+    # @category = Category.find(params[:cat])
```
Comment (Dipeshtwis) [inline app/controllers/entities_controller.rb]: - [x] Here as well. I would recommend removing all the commented code to make the codebase cleaner.

File: app/assets/stylesheets/categories.scss [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+.header-space {
+  padding-right: 3rem;
+}
+
+.header-title {
+  font-size: 25px;
+  font-weight: 300;
+  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
```
Comment (Dipeshtwis) [inline app/assets/stylesheets/categories.scss]: - [x] As per the requirement, the font family should be `Proxima Nova Bold` or `Proxima Nova Light`. Please update to follow the font face requirement.

File: app/views/entities/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+      </div>
+
+      <div class='field'>
+        <%= form.label :amount, 'Amount($)' %>
+        <%= form.number_field :amount, class: 'input' %>
+      </div>
+
+      <div class='field'>
+         <%= form.label :cat, 'Select a category' %>
+          <%# <%= form.collection_select :cat, @categories ,:id , :name %>
```
Comment (Dipeshtwis) [inline app/views/entities/new.html.erb]: - [x] Here as well. I would recommend removing all the commented code to make the codebase cleaner.

File: app/models/category_with_entity.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  belongs_to :entity
+
+  after_validation :update_total_amount
+
+  def update_total_amount
+    cat = Category.find(category_id)
+    ent = Entity.find(entity_id)
+    cat.update(total_amount: cat.total_amount + ent.amount)
+    # cat.total_amount = category.total_amount + ent.amount
+    # cat.save
```
Comment (Dipeshtwis) [inline app/models/category_with_entity.rb]: - [x] Here as well. I would recommend removing the commented code to make the codebase cleaner.

File: README.md [hunk 9]
```diff
@@ -1,24 +1,111 @@
...
+## License
+
+[MIT](./LICENSE)
```
Comment (Dipeshtwis) [inline README.md]: - [x] The Licence link is broken. Please fix that to make the README professional.

File: app/views/devise/sessions/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+        <div class="checkbox">
+          <%= f.check_box :remember_me %>
+          <%= f.label :remember_me %>
+        </div>
+      <% end %>
+
+      <div class="actions">
+        <%= f.submit "Log in", class:'create-btn' %>
+      </div>
+    <% end %>
```
Comment (Dipeshtwis) [inline app/views/devise/sessions/new.html.erb]: - [x] I would recommend to follow the design and color same as given [here](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=).

### Your design

Screenshot from 2021-12-04 17-48-26

### GIven design 

Screenshot from 2021-12-04 17-51-38

File: app/views/devise/registrations/new.html.erb [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+      <div class="field">
+        <%= f.label :name %><br />
+        <%= f.text_field :name, autofocus: true, autocomplete: "name", class: 'input' %>
+      </div>
+
+      <div class="field">
+        <%= f.label :email %><br />
+        <%= f.email_field :email, autofocus: true, autocomplete: "email", class: 'input' %>
+      </div>
+
```
Comment (Dipeshtwis) [inline app/views/devise/registrations/new.html.erb]: - [x] Same with registration page. Please follow the [given design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=). The input field is different and that's okay but keep design same.

### Your design

Screenshot from 2021-12-04 17-54-25



### Given design

Screenshot from 2021-12-04 17-51-11

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @m05e5, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: The application is working fine. Well done.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Emmyn5600 — APPROVED** [review]
Hello,   @m05e5 👋🏾 👋🏾 congratulations on making it all the project requirements.

Highlights:

- Database is well structured 
- Your project is running successfully
- All tests are passing successfully.

## Optional suggestions

- N/a

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

**Pull request merged**